quic/s2n-quic-transport/src/recovery/manager.rs

```diff
@@ -304,6 +305,37 @@ impl<Config: endpoint::Config> Manager<Config> {
         self.pto.on_transmit(context)
     }
 
+    /// Process pending ACK information.
+    ///
+    /// Update congestion controller, timers and meta data around acked packet ranges.
+    pub fn on_pending_ack_ranges<Ctx: Context<Config>, Pub: event::ConnectionPublisher>(
+        &mut self,
+        timestamp: Timestamp,
+        pending_ack_ranges: &mut PendingAckRanges,
+        context: &mut Ctx,
+        publisher: &mut Pub,
+    ) -> Result<(), transport::Error> {
+        debug_assert!(
+            !pending_ack_ranges.is_empty(),
+            "pending_ack_ranges should be non-empty since connection indicated ack interest"
+        );
+
+        let largest_acked_packet_number = pending_ack_ranges
+            .max_value()
+            .expect("pending range should not be empty");
+        self.process_acks(
+            timestamp,
+            pending_ack_ranges.iter(),
+            largest_acked_packet_number,
+            pending_ack_ranges.ack_delay(),
+            pending_ack_ranges.ecn_counts(),
+            context,
+            publisher,
+        )?;
```

Comment:
We need to clear the pending ack ranges after processing them, right?

Reply:
yep. enabling/hooking up the delayed ack processing will happen in a following PR https://github.com/aws/s2n-quic/pull/1294

Reply:
well we still need to clear it after this function is called

Reply:
```suggestion
        )?;
        
        pending_ack_ranges.clear();
```

Reply:
So initially I was hoping to handle it at a higher level (space/application) but I think that could lead to bugs if someone forgot to call the function.

I have captured it in https://github.com/aws/s2n-quic/pull/1293/commits/e25b44e53e41ceba8045e0bd5564c01f26ac25cb

One thing to note is that we clear the queue regardless of success or failure. I think this is right behavior since I dont think we can recover from this.